eng/scripts/Get-TypeSpec-Folders.ps1

```diff
@@ -26,13 +28,15 @@ else {
   }
   Write-Host
 
-  $engFiles = $diffFiles | Where-Object {$_.StartsWith('eng')}
-  $repoRootFiles = $diffFiles | Where-Object {$_ -notmatch [Regex]::Escape([IO.Path]::DirectorySeparatorChar)}
+  $engFiles = $changedFiles | Where-Object {if ($_) { $_.StartsWith('eng') }}
```

Comment:
Are there cases where $_ is empty/null?

Reply:
Yes, any of the null checks in this PR were added to fix bugs we found in testing.  It might be possible to refactor/rewrite the powershell code to be cleaner, but we wanted to get this merged in and working quickly.

Reply:
I believe `$_` was empty/null specifically in the case where there was no git diff (e.g. a PR with no changes).